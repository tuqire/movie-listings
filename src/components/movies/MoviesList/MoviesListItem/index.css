.movies-list-item h4 {
  min-height: 56px;
}

.movies-list-item h6 {
  line-height: 1.5;
  min-height: 38px;
}
