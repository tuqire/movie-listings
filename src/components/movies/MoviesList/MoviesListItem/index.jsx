import React from 'react'
import './index.css'

export default ({ movie }) =>
  <div className="movies-list-item">
    <h4>{movie.title}</h4>
    <h6 className="h6 text-muted small">Genres: {movie.genres.join(', ')}</h6>
    <img className="img-fluid mt-3" src={`${process.env.IMAGE_BASE_URL}/w500${movie.posterPath}`} alt={`poster for ${movie.title}`}/>
  </div>
