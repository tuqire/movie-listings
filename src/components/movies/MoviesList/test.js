import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MoviesList from './index'

Enzyme.configure({ adapter: new Adapter() })

const mockProps = {
  movies: [{
      id: 1,
      genres: ['foo genre', 'bar genre'],
      title: 'movie foo',
      voteAverage: 5,
      genreIds: [250, 251],
      posterPath: '/posterPath-1'
    }, {
      id: 2,
      genres: ['bar genre'],
      title: 'movie bar',
      voteAverage: 6,
      genreIds: [250],
      posterPath: '/posterPath-2'
    }, {
      id: 3,
      genres: ['bar genre', 'foobar genre'],
      title: 'movie foobar',
      voteAverage: 8,
      genreIds: [240],
      posterPath: '/posterPath-3'
    }
  ]
}

describe('components/movies/MoviesList', () => {
  describe('#MoviesList', () => {
    describe('matches snapshots', () => {
      it('when nothing in store', () => {
        const component = mount(<MoviesList movies={[]} />)
        expect(component).toMatchSnapshot()
      })

      it('when movies in store', () => {
        const component = mount(<MoviesList {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })
  })
})
