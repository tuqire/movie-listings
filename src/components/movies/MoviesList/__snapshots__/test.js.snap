// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/movies/MoviesList #MoviesList matches snapshots when movies in store 1`] = `
<Component
  movies={
    Array [
      Object {
        "genreIds": Array [
          250,
          251,
        ],
        "genres": Array [
          "foo genre",
          "bar genre",
        ],
        "id": 1,
        "posterPath": "/posterPath-1",
        "title": "movie foo",
        "voteAverage": 5,
      },
      Object {
        "genreIds": Array [
          250,
        ],
        "genres": Array [
          "bar genre",
        ],
        "id": 2,
        "posterPath": "/posterPath-2",
        "title": "movie bar",
        "voteAverage": 6,
      },
      Object {
        "genreIds": Array [
          240,
        ],
        "genres": Array [
          "bar genre",
          "foobar genre",
        ],
        "id": 3,
        "posterPath": "/posterPath-3",
        "title": "movie foobar",
        "voteAverage": 8,
      },
    ]
  }
>
  <div
    className="row"
  >
    <div
      className="col-6 col-md-4 col-lg-3 mt-5 mb-3"
      key="1"
    >
      <Component
        movie={
          Object {
            "genreIds": Array [
              250,
              251,
            ],
            "genres": Array [
              "foo genre",
              "bar genre",
            ],
            "id": 1,
            "posterPath": "/posterPath-1",
            "title": "movie foo",
            "voteAverage": 5,
          }
        }
      >
        <div
          className="movies-list-item"
        >
          <h4>
            movie foo
          </h4>
          <h6
            className="h6 text-muted small"
          >
            Genres: 
            foo genre, bar genre
          </h6>
          <img
            alt="poster for movie foo"
            className="img-fluid mt-3"
            src="https://image.tmdb.org/t/p/w500/posterPath-1"
          />
        </div>
      </Component>
    </div>
    <div
      className="col-6 col-md-4 col-lg-3 mt-5 mb-3"
      key="2"
    >
      <Component
        movie={
          Object {
            "genreIds": Array [
              250,
            ],
            "genres": Array [
              "bar genre",
            ],
            "id": 2,
            "posterPath": "/posterPath-2",
            "title": "movie bar",
            "voteAverage": 6,
          }
        }
      >
        <div
          className="movies-list-item"
        >
          <h4>
            movie bar
          </h4>
          <h6
            className="h6 text-muted small"
          >
            Genres: 
            bar genre
          </h6>
          <img
            alt="poster for movie bar"
            className="img-fluid mt-3"
            src="https://image.tmdb.org/t/p/w500/posterPath-2"
          />
        </div>
      </Component>
    </div>
    <div
      className="col-6 col-md-4 col-lg-3 mt-5 mb-3"
      key="3"
    >
      <Component
        movie={
          Object {
            "genreIds": Array [
              240,
            ],
            "genres": Array [
              "bar genre",
              "foobar genre",
            ],
            "id": 3,
            "posterPath": "/posterPath-3",
            "title": "movie foobar",
            "voteAverage": 8,
          }
        }
      >
        <div
          className="movies-list-item"
        >
          <h4>
            movie foobar
          </h4>
          <h6
            className="h6 text-muted small"
          >
            Genres: 
            bar genre, foobar genre
          </h6>
          <img
            alt="poster for movie foobar"
            className="img-fluid mt-3"
            src="https://image.tmdb.org/t/p/w500/posterPath-3"
          />
        </div>
      </Component>
    </div>
  </div>
</Component>
`;

exports[`components/movies/MoviesList #MoviesList matches snapshots when nothing in store 1`] = `
<Component
  movies={Array []}
>
  <div
    className="row"
  />
</Component>
`;
